### PR TITLE
Issues/116 crash on multiple set time out calls

### LINKED ISSUE #116: Multiple calls to SetTimeout on the Deepgram client throws exceptions
## What is the current behavior?

When recreating the client and setting the timeout more than once(on different DeepgramClient instances), all calls after the first result in an exception.

## Steps to reproduce

. Create a DeepgramClient
. Set the timeout to any value
. Dispose of the DeepgramClient
. Create a new DeepgramClient
. Set the timeout to any value that is not the same as in step 2.

## Expected behavior

The timeout for the network call should be changed and no exception should be thrown

## Please tell us about your environment

This error is generic and is not tied to any one platform. If it runs .NET, it is affected.

### PR DESCRIPTION
**Pull request by ThindalTV**
Fixes #116.

Fixes the issue where a static and shared HttpClient was passed around all instances of DeepgramClient, no matter if it was still in memory or not(as the HttpClient was still around anyways). This resulted in any calls to SetTimeout after the first request had been sent from ANY DeepgramClient to fail with an exception.

This also sets up recreation of the HttpClient used inside of the DeepgramClient instance when the timeout changes, so that it can be reset to something else, on a per client basis.

---

Basically, this makes each DeepgramClient instance independent from each other, and makes it possible to modify the timeout even after the client has been used to send & recieve the first requests.

### CODE REVIEW

**Comment by CopperBeardy** [comment]
seem good

**Review by CopperBeardy — COMMENTED** [review]
good solution to the issue

**Comment by lukeocodes** [comment]
is this a breaking change for users?

**Comment by CopperBeardy** [comment]
> is this a breaking change for users?

none of the changes users will see
